backend/workers/lib/vod_converter/kitti.py

```diff
@@ -0,0 +1,178 @@
+                     contains left, top, right, bottom pixel coordinates
+   3    dimensions   3D object dimensions: height, width, length (in meters)
+   3    location     3D object location x,y,z in camera coordinates (in meters)
+   1    rotation_y   Rotation ry around Y-axis in camera coordinates [-pi..pi]
+   1    score        Only for results: Float, indicating confidence in
+                     detection, needed for p/r curves, higher is better.
+
+
+"""
+
+import csv
+import os
+import shutil
+
+from PIL import Image
+
+from .abstract import Ingestor, Egestor
+from .labels_and_aliases import output_labels
+
+
+class KITTIIngestor(Ingestor):
+    def validate(self, path, folder_names):
+        expected_dirs = [
+            'images',
+            'labels'
+        ]
+        for subdir in expected_dirs:
+            if not os.path.isdir(f"{path}/{subdir}"):
+                return False, f"Expected subdirectory {subdir} within {path}"
+        if not os.path.isfile(f"{path}/train.txt"):
+            return False, f"Expected train.txt file within {path}"
+        return True, None
+
+    def ingest(self, path, folder_names):
+        image_ids = self._get_image_ids(path)
+        image_ext = 'png'
+        if len(image_ids):
+            first_image_id = image_ids[0]
+            image_ext = self.find_image_ext(path, first_image_id)
+        tmp = [self._get_image_detection(path, image_name, image_ext=image_ext, folder_names=folder_names) for image_name in image_ids]
+        print('size: ' + str(len(tmp)))
+        return tmp
+
+    def find_image_ext(self, root, image_id):
+        for image_ext in ['png', 'jpg']:
+            if os.path.exists(f"{root}/images/{image_id}.{image_ext}"):
+                return image_ext
+        raise Exception(f"could not find jpg or png for {image_id} at {root}/images")
+
+    def _get_image_ids(self, root):
+        path = f"{root}/train.txt"
+        with open(path) as f:
+            return f.read().strip().split('\n')
+
+    def _get_image_detection(self, root, image_id, *, image_ext='png', folder_names):
+        try:
+            detections_fpath = f"{root}/labels/{image_id}.txt"
+            detections = self._get_detections(detections_fpath, image_id)
+            detections = [det for det in detections if det['left'] < det['right'] and det['top'] < det['bottom']]
+            image_path = f"{root}/images/{image_id}.{image_ext}"
+            image_width, image_height = _image_dimensions(image_path)
+            return {
+                'image': {
+                    'id': image_id,
+                    'dataset_id': None,
+                    'path': image_path,
+                    'segmented_path': None,
+                    'width': image_width,
+                    'height': image_height,
+                    'file_name': image_id + '.' + image_ext
+                },
+                'detections': detections
+            }
+        except Exception as e:
+            print(e)
```

Comment:
use logger

Reply:
sys.stdout was temporarily redirected to task.info to omit the need for replacing all prints in new scripts